components/TextInput.js

```diff
@@ -22,12 +22,13 @@ class TextInput extends PureComponent {
 
   handleBlur() {
     this.setState({ isFocused: false });
+    this.props.onBlur()
   }
 
   handleFocus() {
     this.setState({ isFocused: true });
+    this.props.onFocus()
```

Comment:
Destructure `onFocus` from `this.props` on the first line of the method with a fallback to `null` if it isn't set. This should be followed by a newline. Then check if `onFocus` is truthy and run it:

```
handleFocus() {
  const { onFocus = null } = this.props;

  this.setState({ isFocused: true });
  if (onFocus) {
    onFocus();
  }
}
```